### DISCUSSION TITLE
Artifactupload>Lern and ManagedLern are failing

### DISCUSSION THREAD

**Discussion by tsprasath** [Sunbird-Ed/Community#276] [Installation] [answered]
Hi Team

I have been installing sunbird in the GCP environment.The below jobs are failing

**LERN**
**LernDataProduct**

`Started by user [Prasath Sivasubramaniyan](http://localhost:8080/user/prasathsivasubramanian)
Checking out git https://github.com/Sunbird-Lern/data-products into /var/lib/jenkins/workspace/ArtifactUpload/dev/Lern/LernDataProducts@script/633799af93ee86a54d86cc7c2c2199467f2dda9d038e7437ab7edfd89e5433d8 to read pipelines/upload/Jenkinsfile
The recommended git tool is: NONE
No credentials specified
Cloning the remote Git repository
Using shallow clone with depth 1
Cloning repository https://github.com/Sunbird-Lern/data-products
 > git init /var/lib/jenkins/workspace/ArtifactUpload/dev/Lern/LernDataProducts@script/633799af93ee86a54d86cc7c2c2199467f2dda9d038e7437ab7edfd89e5433d8 # timeout=10
Fetching upstream changes from https://github.com/Sunbird-Lern/data-products
 > git --version # timeout=10
 > git --version # 'git version 2.17.1'
 > git fetch --tags --progress --depth=1 -- https://github.com/Sunbird-Lern/data-products +refs/heads/*:refs/remotes/origin/* # timeout=10
 > git config remote.origin.url https://github.com/Sunbird-Lern/data-products # timeout=10
 > git config --add remote.origin.fetch +refs/heads/*:refs/remotes/origin/* # timeout=10
Avoid second fetch
 > git rev-parse origin/release-5.1.0^{commit} # timeout=10
 > git rev-parse release-5.1.0^{commit} # timeout=10
ERROR: Couldn't find any revision to build. Verify the repository and branch configuration for this job.
ERROR: Maximum checkout retry attempts reached, aborting
Finished: FAILURE`

**LernFlinkJobs**
`Started by upstream project "[Build/Lern/LernFlinkJobs](http://localhost:8080/job/Build/job/Lern/job/LernFlinkJobs/)" build number [2](http://localhost:8080/job/Build/job/Lern/job/LernFlinkJobs/2)
originally caused by:
 Started by user [Prasath Sivasubramaniyan](http://localhost:8080/user/prasathsivasubramanian)
Checking out git https://github.com/Sunbird-Lern/data-pipeline.git into /var/lib/jenkins/workspace/ArtifactUpload/dev/Lern/LernFlinkJobs@script/f91fb543787d25fbd0b8c019e835dce1e18ba041fcc57de21cf95502c4ea3b43 to read kubernetes/pipelines/upload/Jenkinsfile
The recommended git tool is: NONE
No credentials specified
Cloning the remote Git repository
Using shallow clone with depth 1
Cloning repository https://github.com/Sunbird-Lern/data-pipeline.git
 > git init /var/lib/jenkins/workspace/ArtifactUpload/dev/Lern/LernFlinkJobs@script/f91fb543787d25fbd0b8c019e835dce1e18ba041fcc57de21cf95502c4ea3b43 # timeout=10
Fetching upstream changes from https://github.com/Sunbird-Lern/data-pipeline.git
 > git --version # timeout=10
 > git --version # 'git version 2.17.1'
 > git fetch --tags --progress --depth=1 -- https://github.com/Sunbird-Lern/data-pipeline.git +refs/heads/*:refs/remotes/origin/* # timeout=10
 > git config remote.origin.url https://github.com/Sunbird-Lern/data-pipeline.git # timeout=10
 > git config --add remote.origin.fetch +refs/heads/*:refs/remotes/origin/* # timeout=10
Avoid second fetch
 > git rev-parse origin/release-5.1.0^{commit} # timeout=10
 > git rev-parse release-5.1.0^{commit} # timeout=10
ERROR: Couldn't find any revision to build. Verify the repository and branch configuration for this job.
ERROR: Maximum checkout retry attempts reached, aborting
Finished: FAILURE`

**ManagedLern**

1. ml-core-service
2. ml-project-service
3. ml-survey-project
4. ml-report-project

`Started by user [Prasath Sivasubramaniyan](http://localhost:8080/user/prasathsivasubramanian)
Replayed [#1](http://localhost:8080/job/ArtifactUpload/job/dev/job/managed-learn/job/ml-core-service/1/)
Loading library deploy-conf@heads/shared-lib
Attempting to resolve heads/shared-lib from remote references...
 > git --version # timeout=10
 > git --version # 'git version 2.17.1'
 > git ls-remote -- https://github.com/project-sunbird/sunbird-devops.git # timeout=10
The recommended git tool is: NONE
No credentials specified
 > git rev-parse --resolve-git-dir /var/lib/jenkins/workspace/ArtifactUpload/dev/managed-learn/ml-core-service@libs/766998390b8294d46fce8cee674175ae640f06f0365a13725b56ecd3d46b6b1c/.git # timeout=10
Fetching changes from the remote Git repository
 > git config remote.origin.url https://github.com/project-sunbird/sunbird-devops.git # timeout=10
Using shallow fetch with depth 1
Fetching upstream changes from https://github.com/project-sunbird/sunbird-devops.git
 > git --version # timeout=10
 > git --version # 'git version 2.17.1'
 > git fetch --no-tags --progress --depth=1 -- https://github.com/project-sunbird/sunbird-devops.git +refs/heads/*:refs/remotes/origin/* refs/heads/shared-lib # timeout=10
Checking out Revision 778da7ce7f97efd003eed98071921a837d209343 (778da7ce7f97efd003eed98071921a837d209343)
 > git config core.sparsecheckout # timeout=10
 > git checkout -f 778da7ce7f97efd003eed98071921a837d209343 # timeout=10
Commit message: "fix: move variables to be accessible across both try blocks"
 > git rev-list --no-walk 778da7ce7f97efd003eed98071921a837d209343 # timeout=10
[Pipeline] Start of Pipeline
[Pipeline] node
Running on [Jenkins](http://localhost:8080/computer/(built-in)/) in /var/lib/jenkins/workspace/ArtifactUpload/dev/managed-learn/ml-core-service
[Pipeline] {
[Pipeline] stage
[Pipeline] { (checkout public repo)
[Pipeline] echo
Found .git folder. Clearing it..
[Pipeline] sh
+ git clean -fxd
Removing metadata.json
Skipping repository private/
Removing private@tmp/
[Pipeline] checkout
The recommended git tool is: NONE
No credentials specified
 > git rev-parse --resolve-git-dir /var/lib/jenkins/workspace/ArtifactUpload/dev/managed-learn/ml-core-service/.git # timeout=10
Fetching changes from the remote Git repository
 > git config remote.origin.url https://github.com/project-sunbird/sunbird-devops.git # timeout=10
Using shallow fetch with depth 1
Fetching upstream changes from https://github.com/project-sunbird/sunbird-devops.git
 > git --version # timeout=10
 > git --version # 'git version 2.17.1'
 > git fetch --tags --progress --depth=1 -- https://github.com/project-sunbird/sunbird-devops.git +refs/heads/*:refs/remotes/origin/* # timeout=10
 > git rev-parse origin/release-5.1.0^{commit} # timeout=10
Checking out Revision 7dc044d7f379ce6cd8289ca5e3d9e54898ff0725 (origin/release-5.1.0)
 > git config core.sparsecheckout # timeout=10
 > git checkout -f 7dc044d7f379ce6cd8289ca5e3d9e54898ff0725 # timeout=10
Commit message: "Merge pull request #3604 from shikshalokam/release-5.1.0"
First time build. Skipping changelog.
[Pipeline] }
[Pipeline] // stage
[Pipeline] stage
[Pipeline] { (Artifact Upload)
[Pipeline] ansiColor
[Pipeline] {

[Pipeline] sh
+ echo ArtifactUpload/dev/managed-learn/ml-core-service
[Pipeline] sh
+ echo ArtifactUpload/dev/managed-learn/ml-core-service
[Pipeline] sh
+ echo ArtifactUpload/dev/managed-learn/ml-core-service
[Pipeline] stage
[Pipeline] { (parameter checks)
[Pipeline] echo
Found environment variable named hub_org with value as: asia.gcr.io/upsmf-368011
[Pipeline] copyArtifacts
Copied 1 artifact from "[Build » managed-learn » ml-core-service](http://localhost:8080/job/Build/job/managed-learn/job/ml-core-service/)" build number [2](http://localhost:8080/job/Build/job/managed-learn/job/ml-core-service/2/)
[Pipeline] sh
+ jq -r .image_name metadata.json
[Pipeline] sh
+ jq -r .node_name metadata.json
[Pipeline] echo
image_tag not specified, using the image_tag specified in metadata.json. 
[Pipeline] sh
+ jq -r .image_tag metadata.json
[Pipeline] }
[Pipeline] // stage
[Pipeline] }

[Pipeline] // ansiColor
[Pipeline] sh
+ pwd
[Pipeline] ansiColor
[Pipeline] {

[Pipeline] stage
[Pipeline] { (ansible-run)
[Pipeline] echo
{env=dev, module=managed-learn, jobName=ml-core-service, absolute_job_path=/Build/managed-learn/ml-core-service, agent=built-in, image_name=ml-core-service, image_tag=release-5.1.0_9838b34_2, currentWs=/var/lib/jenkins/workspace/ArtifactUpload/dev/managed-learn/ml-core-service, ansiblePlaybook=/var/lib/jenkins/workspace/ArtifactUpload/dev/managed-learn/ml-core-service/ansible/docker_image_push.yml, ansibleExtraArgs=--extra-vars "hub_org=asia.gcr.io/upsmf-368011 image_name=ml-core-service image_tag=release-5.1.0_9838b34_2" --vault-password-file /var/lib/jenkins/secrets/vault-pass}
[Pipeline] dir
Running in /var/lib/jenkins/workspace/ArtifactUpload/dev/managed-learn/ml-core-service/private
[Pipeline] {
[Pipeline] echo
Info: Branch override is disabled
[Pipeline] checkout
The recommended git tool is: NONE
using credential github-cred
 > git rev-parse --resolve-git-dir /var/lib/jenkins/workspace/ArtifactUpload/dev/managed-learn/ml-core-service/private/.git # timeout=10
Fetching changes from the remote Git repository
 > git config remote.origin.url https://github.com/tsprasath/sunbird-devops-private.git # timeout=10
Using shallow fetch with depth 1
Fetching upstream changes from https://github.com/tsprasath/sunbird-devops-private.git
 > git --version # timeout=10
 > git --version # 'git version 2.17.1'
using GIT_ASKPASS to set credentials 
 > git fetch --no-tags --progress --depth=1 -- https://github.com/tsprasath/sunbird-devops-private.git +refs/heads/*:refs/remotes/origin/* # timeout=10
 > git rev-parse origin/main^{commit} # timeout=10
Checking out Revision 5022a10df1148f2fdda9c527e30a0828b57ce0fb (origin/main)
 > git config core.sparsecheckout # timeout=10
 > git checkout -f 5022a10df1148f2fdda9c527e30a0828b57ce0fb # timeout=10
Commit message: "changes GCP account name"
 > git rev-list --no-walk a599e29c8ee7a754e06ec73a20afc5e2d54db607 # timeout=10
[Pipeline] }
[Pipeline] // dir
[Pipeline] sh
+ rsync -Lkr /var/lib/jenkins/workspace/ArtifactUpload/dev/managed-learn/ml-core-service/private/ansible/inventory/dev/managed-learn/* /var/lib/jenkins/workspace/ArtifactUpload/dev/managed-learn/ml-core-service/ansible/inventory/env/
rsync: change_dir "/var/lib/jenkins/workspace/ArtifactUpload/dev/managed-learn/ml-core-service/private/ansible/inventory/dev/managed-learn" failed: No such file or directory (2)
rsync error: some files/attrs were not transferred (see previous errors) (code 23) at main.c(1196) [sender=3.1.2]
[Pipeline] }
[Pipeline] // stage
[Pipeline] }

[Pipeline] // ansiColor
[Pipeline] }
[Pipeline] // stage
[Pipeline] ansiColor
[Pipeline] {

[Pipeline] stage
[Pipeline] { (slack_notify)
[Pipeline] sh
+ git rev-parse --short HEAD
[Pipeline] sh
+ rev
+ git name-rev --name-only HEAD
+ + rev
cut -d / -f1
[Pipeline] sh
+ echo ArtifactUpload/dev/managed-learn/ml-core-service
[Pipeline] echo
Could not find env specific Slack channel. Check for global slack channel..
[Pipeline] echo
Could not find slack environment variable. Skipping slack notification..
[Pipeline] }
[Pipeline] // stage
[Pipeline] }

[Pipeline] // ansiColor
[Pipeline] ansiColor
[Pipeline] {

[Pipeline] stage
[Pipeline] { (email_notify)
[Pipeline] sh
+ echo ArtifactUpload/dev/managed-learn/ml-core-service
[Pipeline] echo
Could not find env specific email group. Check for global email group..
[Pipeline] echo
Could not find global email group variable. Skipping email notification..
[Pipeline] }
[Pipeline] // stage
[Pipeline] }

[Pipeline] // ansiColor
[Pipeline] }
[Pipeline] // node
[Pipeline] End of Pipeline
ERROR: script returned exit code 23
Finished: FAILURE`

**Comment by beepdot** [comment, marked as answer]
Do you require managed learn module? Are these jobs mentioned in the documentation? If yes, please share me the link.

**Reply by tsprasath** [reply]
It is not there in the document

**Reply by beepdot** [reply]
Then why are you running these?

**Reply by tsprasath** [reply]
@keshavprasadms not required as part for Sunbird ED setup.Please close this thread
